[Anthony Hermanto] iP

thony-ui opened the pull request:
# Mental Health Chatbot that Keeps Track of Your Tasks!

> “The only journey is the journey within.” – Rainer Maria Rilke ([source](https://americanbehavioralclinics.com/10-motivational-quotes-on-the-importance-of-mental-wellbeing/))

This mental health chatbot will keep track of your to-dos! It is:

- **Text-based**
- **Easy to learn**
- **FAST** – SUPER FAST. Just add whatever tasks you want.

Here is the step-by-step guide to getting the app:

1. [Go to the source code](https://github.com/thony-ui/ip).
2. Click on **Releases**.
3. Download the JAR file.
4. Open your favorite IDE (we recommend IntelliJ).
5. Open the JAR file and run the code 🎉.

And the best part? It’s **FREE**! 😊

## Features:

- [x] **Managing tasks**
- [x] **Managing deadlines** (coming soon)
- [x] **Reminders** (coming soon)
- [ ] **Mental Support** (coming soon)

You can also practice Java using the MentalHealth bot! Here is a snippet of the entry program:

```java
public class Main {
    public static void main(String[] args) {
        new MentalHealth("./data/duke.txt").run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

thony-ui committed:
completed level 0 requirements

thony-ui committed:
Amended Level 0 requirements

thony-ui committed:
Amending Level 0 requirements

thony-ui committed:
Amending level 0 requirements

thony-ui committed:
level-0 completed

thony-ui committed:
level 1 completed

thony-ui committed:
fix level 1 bugs

thony-ui committed:
Level 2 completed

thony-ui committed:
level 3 complete

thony-ui committed:
Level-4

thony-ui committed:
Automated testing

thony-ui committed:
Level-5

thony-ui committed:
Level-6

thony-ui committed:
Added enum for constants

thony-ui committed:
fix automated testing

thony-ui committed:
Level 7 implemented

thony-ui committed:
Merge branch-Level-7 into master to complete Level 7 implementation

thony-ui committed:
Level 8 completed

thony-ui committed:
Merge branch-Level-8 into master to complete Level 8 implementation

thony-ui committed:
A-OOP

thony-ui committed:
Make it more OOP

thony-ui committed:
Merge remote-tracking branch 'origin/add-gradle-support'
merging gradle support

thony-ui committed:
configured gradle

thony-ui committed:
add j unit test case

thony-ui committed:
Add java doc documentation

thony-ui committed:
java coding standards

thony-ui committed:
Merge branch-A-CodingStandard into master

thony-ui committed:
Level-9

thony-ui committed:
package into duke

thony-ui committed:
follow limit on line length

thony-ui committed:
write more javadocs

thony-ui committed:
write more javadocs

thony-ui committed:
coding standard

thony-ui committed:
Implement A-CheckStyle

thony-ui committed:
Merge branch-A-CheckStyle with merge commit

thony-ui committed:
Implement Level-10

thony-ui committed:
Merge branch-Level-10 with merge commit

thony-ui committed:
Bug fixes for java-fx

thony-ui committed:
Fix bugs and customize the user interface

thony-ui committed:
Fix style error in the respective classes

gnoossk commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,317 @@
+     * @return The result of executing the command as a {@code String}.
+     * @throws MentalHealthException If an error occurs while processing the command.
+     */
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws MentalHealthException {
```
Good usage of case and switch!

thony-ui committed:
Add assertions to the command file, ensuring that the relevant methods checks for assertions before executing

thony-ui committed:
Add code quality improvements to the command file, making sure it follows the code quality standards

thony-ui committed:
Merge pull request #2 from thony-ui/branch-A-Assertions

Add assertions to the command file, ensuring that the relevant method…

thony-ui committed:
Merge master into branch-A-CodeQuality to sync with latest changes

thony-ui committed:
Merge pull request #1 from thony-ui/branch-A-CodeQuality

Add code quality improvements to the command file, making sure it fol…

thony-ui committed:
Add yaml file for CI

lutfir-cpu commented on src/main/java/duke/MentalHealth.java:
```diff
@@ -0,0 +1,63 @@
+     */
+    public String getResponse(String input) {
+        ui.greeting();
+        boolean isExit = false;
```
Boolean variables follows naming convention. Good Job!

thony-ui committed:
Merge pull request #3 from thony-ui/A-CI

Add yaml file for CI

lutfir-cpu commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+                String typeOfTask = parts[0];
+                boolean isTaskDone = parts[1].equals("1");
+                String descriptionOfTask = parts[2];
+                IndividualTask task = switch (typeOfTask) {
```
Personally, I would use the switch as follows:

```
IndividualTask task = null;

switch (typeOfTask) {
case "T":
task = new Todo (descriptionOfTask);
...
```

lutfir-cpu commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+     * @return The formatted message with separators.
+     */
+    public String formatMessage(String msg) {
+        return SEPARATOR + "\n" + msg + "\n" + SEPARATOR;
```
I think it could be clearer if we used 'message' for the 'msg' variable.

lutfir-cpu commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+        var db = new DialogBox(text, img);
+        db.flip();
+        return db;
+    }
```
Getter methods follows naming conventions. Good Job!

lutfir-cpu commented:
LGTM! Nitpicking some naming conventions and coding standards but otherwise really nice job.

thony-ui committed:
Amend CI pipeline to include test cases and improve code quality

The previous CI pipeline included unnecessary deployment steps that were not required for this project. Additionally, test cases were not being run as part of the pipeline, which could lead to code being merged without proper validation.

This commit addresses these issues by:
* Removing the unnecessary deployment steps from the CI pipeline.
* Adding test execution as part of the build process to ensure code   passes all tests before being merged.
* Enhancing the code base by improving the overall code quality,   addressing readability, and ensuring consistency across the codebase.

Let's ensure that code quality is maintained by integrating tests into the CI pipeline and removing redundant processes. This will improve the reliability of the code and streamline the development workflow.

By making these changes, we enhance the efficiency of the CI pipeline and ensure only properly tested code can be merged.

thony-ui committed:
Improve code quality through refactoring and simplification

The current codebase contains redundant code blocks, deep nesting, and some complex expressions, which reduce readability and maintainability.

To improve code quality, this commit performs the following:
* Simplifies conditional statements by reducing deep nesting.
* Refactors repetitive code into reusable methods, adhering to DRY principle.
* Removes dead code and unnecessary comments to improve clarity.
* Introduces better variable naming for improved readability.

These changes enhance the maintainability of the codebase, making it easier to understand and modify in the future while reducing potential for errors.

thony-ui committed:
Support ability to edit task details without deletion

Currently, the application requires users to delete and recreate a
task to make changes, which is not user-friendly and time-consuming.
Editing details like the end time of an event or description of a task
should not require complete removal of the task.

To address this, I have implemented a feature allowing users to update
specific fields of `ToDo`, `Deadline`, and `Event` tasks. Now, users can
modify details such as the description, deadline, or event times without
having to delete the entire task.

Key changes:
* Added a method `handleUpdateCommand` to handle update logic.
* Supported the ability to update fields like description, 'by' (deadline),
  and event start/end times using field-specific update methods.
* Ensured that multi-word values can be provided using string concatenation
  logic for the update command.

This improvement enhances usability, allowing more efficient task management
without unnecessary task deletion and re-creation.

thony-ui committed:
Merge pull request #4 from thony-ui/branch-C-Update

Support ability to edit task details without deletion

thony-ui committed:
Code quality changes to all the files

thony-ui committed:
Add streams to TaskList class

thony-ui committed:
Add streams to Command class

thony-ui committed:
Code quality fixes

thony-ui committed:
Fix the issue when data directory or duke file is not found

thony-ui committed:
Add pre-commit hook to ensure test cases pass

thony-ui committed:
Modify A-CI to also test for style, ensuring no style violations

thony-ui committed:
Add pre-commit hooks and modify CI file

thony-ui committed:
Final changes for production

thony-ui committed:
Add image to the product page and modify the product website

thony-ui committed:
Bug fixes for the product page

thony-ui committed:
Modify the readme file to ensure that the link to release is working

thony-ui committed:
Fix bug related problems when conducting stress checks

thony-ui committed:
Fix readme so that the expected output matches

thony-ui committed:
Update readme file to ensure credits are being shown
